src/components/SearchForm.vue

```diff
@@ -107,11 +108,17 @@ export default {
     isNewAdultIncluded() {
       return this.adultIncluded !== this.booleanQueryAdultIncluded
     },
+    isNewPage() {
+      return this.page !== +this.$route.query.page
+    },
     isNewSearch() {
-      return this.isNewSearchValue || this.isNewAdultIncluded
+      return this.isNewSearchValue || this.isNewAdultIncluded || this.isNewPage
```

Comment:
i don't think its a good idea to manualy check every parametr and this part need to be changed